### PR TITLE
feat: add type to locales and remove automatic mapping

### PR DESCRIPTION
**Pull request by alexandre-combemorel**

### CODE REVIEW

File: packages/use-i18n/src/__tests__/usei18n.test.tsx [hunk 1]
```diff
@@ -20,7 +22,8 @@ type NamespaceLocale = {
 
 type OnTranslateError = ComponentProps<typeof I18n>['onTranslateError']
 
-const defaultSupportedLocales = ['en', 'fr', 'es']
+const isLocaleSupportedDefault = (locale: string): locale is Locales =>
```
Comment (philibea) [inline packages/use-i18n/src/__tests__/usei18n.test.tsx:25]:
```suggestion
const isDefaultSettingsLocalesSupported = (locale: string): locale is Locales =>
```

sound better ?

File: packages/use-i18n/src/usei18n.tsx [hunk 2]
```diff
@@ -93,8 +84,8 @@ const getCurrentLocale = ({
   return defaultLocale
 }
 
-type Context<Locale extends BaseLocale> = {
-  currentLocale: string
+type Context<LocaleParam extends BaseLocale, T extends string> = {
```
Comment (philibea) [inline packages/use-i18n/src/usei18n.tsx:87]: What is T ?

File: packages/use-i18n/src/__tests__/usei18n.test.tsx (hunk 1 above)
Comment (alexandre-combemorel) [reply packages/use-i18n/src/__tests__/usei18n.test.tsx:25]: Agree, I will just remove settings in the name, as we only check for the list of locales and not the whole settings list of locale

File: packages/use-i18n/src/usei18n.tsx (hunk 2 above)
Comment (alexandre-combemorel) [reply packages/use-i18n/src/usei18n.tsx:87]: T is the union list of locale types supported. I will rename it

**Review by johnrazeur — APPROVED** [review]

File: packages/use-i18n/src/usei18n.tsx [hunk 3]
```diff
@@ -23,53 +23,54 @@ import type { ReactParamsObject, ScopedTranslateFn, TranslateFn } from './types'
...
-        t: (str: 'You must pass a generic argument to useI18n()') => void
-        namespaceTranslation: (
-          str: 'You must pass a generic argument to useI18n()',
-        ) => void
-      }
-    : Context<Locale>
+type RequiredGenericContext<
+  LocaleParam extends BaseLocale,
+  LocalSupportedType extends string,
+> = keyof string extends never
```
Comment (philibea) [inline packages/use-i18n/src/usei18n.tsx:34]:
```suggestion
> = keyof LocaleParam extends never
```

Keyof string extends never will be always false ? We try to check the generic here

**Review by philibea — APPROVED** [review]

File: packages/use-i18n/src/usei18n.tsx (hunk 3 above)
Comment (alexandre-combemorel) [reply packages/use-i18n/src/usei18n.tsx:34]: indeed ! good catch

**Pull request merged**